tests/cache-test/Makefile

```diff
@@ -0,0 +1,52 @@
+include ../../mk/common.mk
+
+.PHONY: all check clean
+
+BUILDDIR := build
+SOURCE := ../../src
+TARGET := testCache
+TESTDIR := testCases
+
+CFLAGS = -std=gnu99 -O2 -Wall -Wextra
+CFLAGS += -Wno-unused-label -I$(SOURCE)
+
+OBJS := cache.o \
+        testCache.o
+
+OBJS := $(addprefix $(BUILDDIR)/, $(OBJS))
+deps := $(OBJS:%.o=%.o.d)
+
+CHECK_ELF_FILES := \
```

Comment:
You may consider to check shecc for its test suite.
See [tests/driver.sh](https://github.com/jserv/shecc/blob/master/tests/driver.sh) and [Makefile](https://github.com/jserv/shecc/blob/master/Makefile).